api/app/services/event_member_service.rb

```diff
@@ -0,0 +1,62 @@
+class EventMemberService
+  extend Sortable
+  extend Paginatable
+
+  SORTABLE_COLUMNS = {
+    **EventService::SORTABLE_COLUMNS,
+    "rating" => :rating
+  }.freeze
+
+  def self.fetch(params, current_user)
+    scope = current_user.event_members.joins(:event)
+    scope = sort(scope, params, SORTABLE_COLUMNS)
+    scope = paginate(scope, params)
+
+    scope.preload(:event)
+  end
+
+  def self.create(event, params, current_user)
+    number_of_tickets = extract_number_of_tickets(params)
+    event_members = []
+
+    ActiveRecord::Base.transaction do
+      event.with_lock do
+        raise Api::Errors::CommonError::InvalidParams.new(id: event.id) unless event.joinable?
+
+        check_tickets_availability(event, number_of_tickets)
+
+        event_members = Array.new(number_of_tickets) do
+          EventMember.new(
+            event: event,
+            user: current_user,
+            ticket_qr_code: SecureRandom.uuid
+          )
+        end
+
+        EventMember.import(event_members)
+      end
+    end
+
+    event_members
+  end
+
+  private
+
+  def self.extract_number_of_tickets(params)
+    num = params[:number_of_tickets].to_i
+    num = 1 if num <= 0
+
+    num
+  end
```

Comment:
The function reads params[:number_of_tickets] but the controller will (should) pass a nested event_member hash (e.g. { event_member: { number_of_tickets: 2 } }). This mismatch leads to defaulting to 1 ticket or incorrect behavior.

Accept the nested shape or change the controller to pass the scalar. For service change:
```
def self.extract_number_of_tickets(params)
  num = if params.is_a?(ActionController::Parameters) && params[:event_member]
          params[:event_member][:number_of_tickets].to_i
        else
          params[:number_of_tickets].to_i
        end
  num = 1 if num <= 0
  num
end
```
Or have the controller call: `EventMemberService.create(@event, event_member_params[:number_of_tickets], current_user)`.